c++/h5/group.cpp

```diff
@@ -95,6 +95,17 @@ namespace h5 {
     return group(obj, parent_file);
   }
 
+  void group::create_softlink(std::string const &target_path, std::string const& link_name) const {
+    if (target_path.empty() || link_name.empty()) return;
+    if (!has_key(target_path)) throw std::runtime_error("The target " + target_path + " does not exist in group " + name());
+    if (has_key(link_name)) {
+        auto const err = H5Ldelete(id, link_name.c_str(), H5P_DEFAULT);
+        if (err < 0) throw std::runtime_error("Cannot delete existing softlink " + link_name);
+    }
```

Comment:
I think we should opt to just throw if the `key` or `link_name` already exists and not to silently delete existing ids.

Reply:
I've implemented this now, but this kind of negates the entire purpose of this, because as far as I can see it is not possible to manually delete items from the `HDFArchive`.

Reply:
It is true that it is currently not possible to delete keys from the archive. In fact I overlooked that for both `create_dataset` and `create_group` functions the default is currently to call `unlink(key)`, i.e. delete the previous `key` if it already existed. We should be doing the same also for `create_softlink` to remain consistent.

Could you make that adjustment? Sorry for the back and forth.